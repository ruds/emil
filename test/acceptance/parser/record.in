{foo=1,bar={baz=quux},waldo=(prefix -)} ;
{};
{foo=1,foo=bar};
{foo:1};
{foo};
