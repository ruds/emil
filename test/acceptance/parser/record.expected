0001 (ExprTopDecl
         :expr
         (RecordExpr
             :rows
             (
                 (RecRowExpr
                     :label
                     "666F6F"
                     :val
                     (BigintLiteralExpr
                         :val
                         1))
                 (RecRowExpr
                     :label
                     "626172"
                     :val
                     (RecordExpr
                         :rows
                         ((RecRowExpr
                             :label
                             "62617A"
                             :val
                             (IdentifierExpr
                                 :qualifiers
                                 ()
                                 :identifier
                                 "71757578"
                                 :is_op
                                 false
                                 :is_prefix_op
                                 false)))))
                 (RecRowExpr
                     :label
                     "77616C646F"
                     :val
                     (IdentifierExpr
                         :qualifiers
                         ()
                         :identifier
                         "2D"
                         :is_op
                         true
                         :is_prefix_op
                         true))
             )))
0002 (ExprTopDecl
         :expr
         (RecordExpr
             :rows
             ()))
0003 ERROR
0004 ERROR
0005 ERROR
0006 (EndOfFileTopDecl)
